executor/distsql.go

```diff
@@ -239,10 +240,46 @@ func (e *IndexReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 		return nil
 	}
 
+	// Global index has the last partitionID column, filter out needed rows and prune the last column.
+	if e.index.Global {
+		req.Reset()
+		retTps := append(e.base().retFieldTypes, types.NewFieldType(mysql.TypeLonglong))
+		if err := e.result.ResetFieldTypes(retTps); err != nil {
+			return err
+		}
+
+		// TODO(L-Maple): chk should be reused
+		chk := chunk.NewChunkWithCapacity(retTps, req.Capacity())
```

Comment:
Would this be needed if the column was already in the output columns?
Any reason for not including the PartitionID (physical table id) in the output, to avoid yet another pruning call if need to read columns not included in the index?

I assume if we add the ExtraPidColID column to the v.OutputColumns that we also would need a column pruning step in the plan?

Reply:
this logic is removed now, as I push down the filter partitionIds.